sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -151,6 +152,41 @@ class UDFSuite extends QueryTest with SharedSQLContext {
     assert(sql("SELECT strLenScala('test', 1)").head().getInt(0) === 5)
   }
 
+  test("Deterministic and literal UDF optimization") {
```

Comment:
Also, could you make a separate test suite for that new optimizer?

Reply:
Thanks @dongjoon-hyun for your comment.  

I wanted to make sure I understood correctly:
1. Did you want me to add this test case in a separate test suite?  
2. Or should I add some more new tests - unit test like for optimizer rule?

Can you let me know.

Reply:
What I meant was (1).

Reply:
Thank you. I have taken care of (1). Moved tests to a new test suite.